### PR TITLE
Search all Posts in Discussion Forum

### PR DESCRIPTION
**Pull request by mumerfarooq000**
Story Links

[COL-125](https://edlyio.atlassian.net/browse/COL-125)

PR Links
[Colaraz Edx-Platform](https://github.com/colaraz/edx-platform/pull/82)

description: 
added search functionality for all the posts done on a particular course which is displayed in the all discussion dropdown.

### CODE REVIEW

File: colaraz/lms/static/js/discussion/colaraz_discussion_filters.js [hunk 1]
```diff
@@ -304,4 +305,63 @@ function reLoadDropDowns() {
...
+
+    e.stopPropagation();
+    e.preventDefault();
+
+});
+
+//reload all dicussions again
+$(document).on('click', '#cz-search-cancel', function (e) {
+    $('#all-discussions-filter').val('');
+    nextUrlAllDiscussions = colarazDiscussionAllThreadsUrl;
```
Comment (danialmalik) [inline colaraz/lms/static/js/discussion/colaraz_discussion_filters.js:353]: Why are we re-assigning the value here?

File: colaraz/lms/static/js/discussion/colaraz_discussion_filters.js [hunk 2]
```diff
@@ -109,7 +110,7 @@ function loadMoreAllTopics(element) {
 
 function loadMoreButtonAllTopics(next_url) {
     return `<li class="forum-nav-thread" style="user-select: auto;">
-                <a id="cz-loadmore" href="#" class="forum-nav-thread-link" next-url="${next_url}" onclick="loadMoreAllTopics(this)">load more</a>
+                <a id="cz-loadmore" href="#" class="forum-nav-thread-link " next-url="${next_url}" onclick="loadMoreAllTopics(this)">load more</a>
```
Comment (imhassantariq) [inline colaraz/lms/static/js/discussion/colaraz_discussion_filters.js:113]: Why are we adding space in class? Is there any specific reason behind this?

File: colaraz/lms/static/js/discussion/colaraz_discussion_filters.js (hunk 1 above)
Comment (mumerfarooq000) [reply colaraz/lms/static/js/discussion/colaraz_discussion_filters.js:353]: when user press load more it gets updated according to pagination so we reset it to its previous state once cancel button is clicked.

File: colaraz/lms/static/js/discussion/colaraz_discussion_filters.js (hunk 2 above)
Comment (mumerfarooq000) [reply colaraz/lms/static/js/discussion/colaraz_discussion_filters.js:113]: typo resolved.

File: colaraz/lms/static/js/discussion/colaraz_discussion_filters.js (hunk 1 above)
Comment (imhassantariq) [reply colaraz/lms/static/js/discussion/colaraz_discussion_filters.js:353]: Okay.

**Pull request merged**